### PR TITLE
Add BL5340 support to ipc samples

### PR DESCRIPTION
**Pull request by nordicjm**
samples: ipc: rpmsg_service, samples: ipc: rpmsg_multi_instance, samples: ipc: ipc_service: Add support for BL5340

Add support for BL5340 to various ipc sample applications

### CODE REVIEW

**Review by cfriedt — APPROVED** [review]
LGTM

File: samples/subsys/ipc/ipc_service/CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,8 @@ set(REMOTE_ZEPHYR_DIR ${CMAKE_CURRENT_BINARY_DIR}/ipc_service_remote-prefix/src/
 
 if("${BOARD}" STREQUAL "nrf5340dk_nrf5340_cpuapp")
   set(BOARD_REMOTE "nrf5340dk_nrf5340_cpunet")
+elseif("${BOARD}" STREQUAL "bl5340_dvk_cpuapp")
+  set(BOARD_REMOTE "bl5340_dvk_cpunet")
```
Comment (tejlmand) [inline samples/subsys/ipc/ipc_service/CMakeLists.txt:13]: I see you're not the one introducing this, but I find this to restrictive in general.

There will be more boards capable of supporting this sample.
Like all `nRF5340` based boards, and most likely also several lpcxpresso boards and some other boards.

We have several samples where there are specific requirements to the hw for running the sample, but we don't filter on boards in those samples, for example the `can/isotp` sample requires CAN bus support, and not all boards have that.

But seems to have been used in most samples in `subsys/ipc`.

**Review by tejlmand — CHANGES_REQUESTED** [review]
In general, I'm not fond of how most samples in `subsys/ipc` is handling boards, but that is not for this PR to fix.

But with the current way subsys/ipc is handled, then you must also update sample docs for:
- IPC Service                                                                                          
- RPMsg multi instance                                                                                 
- RPMsg service      

with the new board.

**Comment by tejlmand** [comment]
and please make a proper PR description.
Makes life much easier for reviewers.

**Review by tejlmand — APPROVED** [review]

**Comment by carlocaione** [comment]
Uhm, just an heads up that this PR is going to be lost in the wind when #38734 will be merged. There is currently an ongoing effort to remove `rpmsg_multi_instance`.

**Pull request merged**